src/snowflake/cli/_plugins/nativeapp/entities/application.py

```diff
@@ -826,7 +827,16 @@ def get_events(
         try:
             return sql_executor.execute_query(query, cursor_class=DictCursor).fetchall()
         except ProgrammingError as err:
-            generic_sql_error_handler(err)
+            if err.errno == DOES_NOT_EXIST_OR_NOT_AUTHORIZED:
+                raise ClickException(
+                    dedent(
+                        f"""\
+                    Event table '{account_event_table}' does not exist or you are not authorized to perform this operation.
+                    Please check your EVENT_TABLE parameter to ensure that it is set to a valid event table."""
+                    )
+                ) from err
```

Comment:
Used a ClickException here since it's a specific case and I didn't want to add to our ever increasing list of custom errors, but also wanted to tune the error message to be a bit more specific/actionable